### ISSUE TITLE
Unable to reply to some contacts, the application incorrectly shows that the contact number is a shortcode 

### ISSUE THREAD

**Issue by JanicePolito** [FossifyOrg/Messages#224]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Android 14

### Affected device model

Motorola edge 50 pro

### How did you install the app?

Google Play Store

### Steps to reproduce the bug

1. Open app
2. Choose affected conversation
<img alt="Screenshot_20240914-152730_Wiadomości" src="https://github.com/user-attachments/assets/d0c72da6-b458-4815-9436-b3289e6f051b" width=179 />
3. Can't send message, the application incorrectly shows that the contact number is a shortcode 

### Expected behavior

The app should be able to send messages to normal contacts. 

### Actual behavior

Unable to reply to some contacts, the application incorrectly shows that the contact number is a shortcode 

### Screenshots/Screen recordings

![Screenshot_20240914-153456_Wiadomości](https://github.com/user-attachments/assets/9e6efaf2-470d-4f57-a034-f0f12946b307)

**Comment by Aga-C** [comment]
1. Were you having this conversation earlier in Fossify Messages, and it just stopped working? Or these are old messages that you exchanged in another app?
2. Can you share how that number is saved in your contacts? I don't mean the exact number, but a pattern like `+48 (00) 000 00 00` or `000 000 000`. I'd like to know if there's something despite digits and how long it is.

**Comment by markbara-git** [comment]
> 1. Were you having this conversation earlier in Fossify Messages, and it just stopped working? Or these are old messages that you exchanged in another app?
> 2. Can you share how that number is saved in your contacts? I don't mean the exact number, but a pattern like `+48 (00) 000 00 00` or `000 000 000`. I'd like to know if there's something despite digits and how long it is.

1. The caller sent me a message on Fossify. I could not reply to it due to the number being mistakenly considered short-code. I held the conversation in another app and now I'm back on the Fossify Messages app.
2. The screenshots below show what format the sender's number is stored in:
![Screenshot_20240916-054350_Wiadomości](https://github.com/user-attachments/assets/10a93b8b-caa2-40e3-8b79-9de5deabbc70)
![Screenshot_20240916-054244_Wiadomości](https://github.com/user-attachments/assets/29d92317-2ca2-4a2b-9e5d-5153dcc49701)
![Screenshot_20240916-054239_Kontakty](https://github.com/user-attachments/assets/cb2d77f2-07fa-4d17-b5ac-d32d88d2e1ee)

**Comment by naveensingh** [comment]
Could not reproduce this after multiple tries on different phones, does anybody else have this problem?

**Comment by alextingle** [comment]
I get this too.

Version 1.0.1
Motorola edge 50 neo.
Android 14.
Obtained from fdroid.

**Comment by naveensingh** [comment]
@alextingle what's your system language?

**Comment by alextingle** [comment]
System language is "English (United Kingdom)".

**Comment by THEKEV46** [comment]
Hi having same problem replying to message sent by Motorola RAZR40 ultra from my new 
Samsung Galaxy Tab A9

![IMG_20250115_140446783](https://github.com/user-attachments/assets/1f68c4c6-abdf-4ab7-9065-ffe208bce37c)

**Comment by Aga-C** [comment]
@JanicePolito @alextingle It may sound strange, but is there a chance that senders recognized as short codes use Motorola phones?

**Comment by THEKEV46** [comment]
Can send reply to razr using google messages. I use fossify messages on razr

**Comment by Aga-C** [comment]
> Can send reply to razr using google messages. I use fossify messages on razr

@THEKEV46 Is it Samsung on the photo with Fossify Messages? If yes, can you send SMS from Fossify Messages there?

Also, do you have the same issue on your Razr?

**Comment by THEKEV46** [comment]
Yes it's a samsung a9 tablet running android 14. razr also running android
14 fossify has no issues on razr. Icon color edit also not working on
samsung a9

**Comment by JanicePolito** [comment]
In the latest version i.e. 1.1.5 the issue seems to have been fixed

**Comment by naveensingh** [comment]
Cool, thanks for the update!

@alextingle @THEKEV46 Do you guys still have this problem?

**Comment by THEKEV46** [comment]
Hi Janice,

I confirmed issue now fixed for my Samsung A9 tablet using version 1.1.5
and the icon colour edit works too. 😀

Thanks
Kev

**Comment by naveensingh** [comment]
I received a report that this is still happening.

**Comment by smschm** [comment]
I'm currently having this problem. I have version 1.8.0, installed from FDroid. Phone is a Pixel 7 with Android 15. The contact that this shows up for has their phone number appear sometimes in +19876543210 format, sometimes +1 (987) 654-3210, depending on where.

**Comment by naveensingh** [comment]
Please check again with v1.8.1. It'll be released in the next hour or so.

**Comment by smschm** [comment]
Yes, this was fixed for me in v1.8.1. Thanks!

**Comment by TheFwuffernaut** [comment]
still having this issue as of 1.9.1.

Worked fine as of July 1st, broke somewhere between then and July 21st, persists to time of posting.

Contact is stored as 01234567891 in contacts, number that messages tries to contact is 1234567891 with dropped 0.
Cannot see where/how to add +44 area code in either contacts or messages the event that this fixes it (contacts does not allow for a + to put in the +44).

Edit 1; found the +, testing now with +44 in place of 0
Edit 2; did not change anything for original conversation, cannot make a "new" chain in the event that this can fix the issue as it defaults to the old chain with the still-missing digits. Don't want to delete the old message chain for obvious reasons.
